File: Integrations/sdks/python/scriptbee_analysis_sdk/domain/context.py
from pydantic import BaseModel


class ContextSlice(BaseModel):
    model: str
    plugin_ids: list[str]


class ContextGraphNode(BaseModel):
    id: str
    label: str
    type: str
    loader: str | None = None
    properties: dict[str, object] = {}


class ContextGraphEdge(BaseModel):
    source: str
    target: str
    label: str


class ContextGraphResult(BaseModel):
    nodes: list[ContextGraphNode]
    edges: list[ContextGraphEdge]

File: Integrations/sdks/typescript/src/domain/context.ts
import { z } from "zod";

export const ContextSliceSchema = z.object({
  model: z.string(),
  pluginIds: z.array(z.string()),
});
export type ContextSlice = z.infer<typeof ContextSliceSchema>;

export const ContextGraphNodeSchema = z.object({
  id: z.string(),
  label: z.string(),
  type: z.string(),
  loader: z.string().nullable().optional(),
  properties: z.record(z.unknown()).default({}),
});
export type ContextGraphNode = z.infer<typeof ContextGraphNodeSchema>;

export const ContextGraphEdgeSchema = z.object({
  source: z.string(),
  target: z.string(),
  label: z.string(),
});
export type ContextGraphEdge = z.infer<typeof ContextGraphEdgeSchema>;

export const ContextGraphResultSchema = z.object({
  nodes: z.array(ContextGraphNodeSchema),
  edges: z.array(ContextGraphEdgeSchema),
});
export type ContextGraphResult = z.infer<typeof ContextGraphResultSchema>;
